### ISSUE TITLE
🐛 BUG: Can't import framework components using relative paths

### ISSUE THREAD

**Issue by AmineAML** [withastro/astro#2286]
### What version of `astro` are you using?

0.21.13

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

I'm trying to use partial hydration like`client:load` for some necessary JavaScript on a component (using Svelte) then this error shows up: 
```[plugin:vite:import-analysis] Failed to resolve import "/a_long_path/src/components/example.svelte" from "src/pages/something.astro?html-proxy&index=0.js". Does the file exist?
1  |  import setup from 'astro/client/load.js';
2  |  setup("1ciFDX", {value: true}, async () => {
3  |    const [{ default: Component }, { default: hydrate }] = await Promise.all([import("/a_long_path/src/components/Example.svelte"), import("@astrojs/renderer-svelte/client.js")]);
   |                                                                                     ^
4  |    return (el, children) => hydrate(el)(Component, {
5  |    "example": [{
```    
I've tried it on an older version of astro and of the svelte renderer plugin and didn't face any issue, so I guess it could be because of changes in the recent update.   

This could be a bug or something from my part like having whitespace in the path causing an issue?

### Link to Minimal Reproducible Example

https://github.com/AmineAML/fadayiy-app

**Comment by jonathantneal** [comment]
Hello! I’m unable to investigate this issue, because the `graphql-request` dependency is throwing this error:

```
Only absolute URLs are supported
```

Would you be able to resolve this for me to investigate, or is this a related issue?

**Link to Minimal Reproducible Example**

https://stackblitz.com/github/AmineAML/fadayiy-app?file=package.json

**Comment by AmineAML** [comment]
@jonathantneal Hey, could you please use this link: https://stackblitz.com/edit/github-pcus5m?file=package.json
The issue you had was because the app was not connecting to the API, also for better debugging the problem in question try to hydrate some components anywhere on the project and let me know what you found.
Note that I cannot reproduce the problem on stackblitz, but locally the problem still persists.

**Comment by RafidMuhymin** [comment]
I'm pretty sure what the bug is. A few users and I have experienced this issue. @AmineAML  I think you have included just a part of the error. I think the complete error is like the below screenshot.

![image](https://user-images.githubusercontent.com/63650415/151167151-c6be4295-0fca-4af4-b6f7-3c8bbdc34c48.png)

If you look at the error you'll see that the first path that Vite is trying to resolve is relative to the current volume of your HDD/SSD. But the second path where the import statement is coming from is project-directory relative. Can you change the import statement of the component you are trying to import from module-relative to project-relative

For example, change this,
```js
import Counter from "../path/to/SvelteCounter.vue";
```
To,
```js
import Counter from "/src/components/SvelteCounter.vue"
```

@jonathantneal @AmineAML You can take a look at the below discord threads.

https://discord.com/channels/830184174198718474/830184175176122389/919523853116538890
https://discord.com/channels/830184174198718474/924222854843748402/924227634810986537
https://discord.com/channels/830184174198718474/915952854408249364/916384575008161892

**Comment by AmineAML** [comment]
@RafidMuhymin I've just got back to this project and you're indeed correct, that's the same issue I was getting and after I've applied your fix is now works fine, I guess this has to do with the Astro update as it works similarly as I had before in another project that still uses an older version of the framework.
Thank you!

**Comment by RafidMuhymin** [comment]
@AmineAML I am reopening the issue as it's a cross-platform compatibility issue and it should be resolved.

**Comment by FredKSchott** [comment]
I believe that this has been fixed, if you can still reproduce in v0.24.0 please let me know!

**Comment by RafidMuhymin** [comment]
Yeah @FredKSchott it has been fixed in the recent releases. I forgot to post an update on this issue. Thank you!
